src/gt4py/next/program_processors/runners/dace_fieldview/transformations/strides.py

```diff
@@ -97,3 +107,125 @@ def _find_toplevel_transients(
+        for edge_from_map_entry in state.out_edges_by_connector(edge.dst, map_entry_out_conn):
+            gt_map_strides_to_dst_nested_sdfg(sdfg, state, edge_from_map_entry, outer_node)
+        return
+
+    if not isinstance(edge.dst, dace.nodes.NestedSDFG):
+        return
+
+    outer_strides = outer_node.desc(sdfg).strides
+    _gt_map_strides_to_nested_sdfg(edge.dst, edge.dst_conn, edge.data, outer_strides)
+
+    for inner_state in edge.dst.sdfg.states():
+        for inner_node in inner_state.data_nodes():
+            if inner_node.data == edge.dst:
+                for inner_edge in inner_state.out_edges(inner_node):
+                    gt_map_strides_to_dst_nested_sdfg(sdfg, state, inner_edge, inner_node)
+
+
+def gt_map_strides_to_src_nested_sdfg(
+    sdfg: dace.SDFG,
+    state: dace.SDFGState,
+    edge: dace.sdfg.graph.Edge,
+    outer_node: dace.nodes.AccessNode,
+) -> None:
+    """Propagates the strides of the given data node to the nested SDFGs on the edge source.
+
+    This function will recursively visit the nested SDFGs connected to the given
+    data node and apply mapping from inner to outer strides.
+
+    Args:
+        sdfg: The SDFG to process.
+        state: The state where the data node is used.
+        edge: The edge that writes to the data node, the nested SDFG is expected as the source.
+        outer_node: The data node whose strides should be propagated.
+    """
+    if isinstance(edge.src, dace.nodes.MapExit):
+        # Find the source of the edge entering the map exit node
+        map_exit_in_conn = edge.src_conn.replace("OUT_", "IN_")
+        for edge_to_map_exit in state.in_edges_by_connector(edge.src, map_exit_in_conn):
+            gt_map_strides_to_src_nested_sdfg(sdfg, state, edge_to_map_exit, outer_node)
+        return
+
+    if not isinstance(edge.src, dace.nodes.NestedSDFG):
+        return
+
+    if isinstance(edge.src.sdfg.data(edge.src_conn), dace.data.Scalar):
+        return  # no strides to propagate
+
+    outer_strides = outer_node.desc(sdfg).strides
+    _gt_map_strides_to_nested_sdfg(edge.src, edge.src_conn, edge.data, outer_strides)
+
+    for inner_state in edge.src.sdfg.states():
+        for inner_node in inner_state.data_nodes():
+            if inner_node.data == edge.src_conn:
+                for inner_edge in inner_state.in_edges(inner_node):
+                    gt_map_strides_to_src_nested_sdfg(sdfg, state, inner_edge, inner_node)
+
+
+def _gt_map_strides_to_nested_sdfg(
+    nsdfg_node: dace.nodes.NestedSDFG,
+    inner_data: str,
+    edge_data: dace.Memlet,
+    outer_strides: Iterable[int | dace.symbolic.SymExpr],
+) -> None:
+    # We need to propagate the strides inside the nested SDFG on the global arrays
+    new_strides = tuple(
+        stride
+        for stride, to_map_size in zip(
+            outer_strides,
+            edge_data.subset.size(),
+            strict=True,
+        )
+        if to_map_size != 1
```

Comment:
@edopao why do we exclude these dimensions?

Reply:
No, actually there was a reason. The reason is that currently the scan is writing into a 1D array along the vertical dimension, so the memlet from the nested SDFG implementing the scan is implicitly un-slicing the data into the outer field (which is usually multidimensional). So that code is there to handle array slicing through memlet subset. Question is whether we want to support it, or not since it will be deprecated in next dace release.